### ISSUE TITLE
iFrame: Sometimes shift+tab does not focus the block toolbar

### ISSUE THREAD

**Issue by alexstine** [WordPress/gutenberg#46959]
### Description

At times, the shift+tab shortcut while in a block does not focus the block toolbar. I believe this is a regression from the iFrame implementation and should be addressed ASAP.

### Step-by-step reproduction instructions

1. Create a new post or page.
2. Focus is placed in the title field inside the iFrame.
3. Give your post/page a title.
4. press enter to create a new block.
5. Start typing to automatically create a paragraph block.
6. Press shift+tab.
7. Notice how focus lands on options button instead of the toolbar for the paragraph block.
8. Press tab to move back into the block.
9. Press alt+f10 to focus the block toolbar.
10. Press tab to go back into the block.
11. Press shift+tab and notice how the toolbar correctly receives focus.

### Environment info

- Gutenberg: trunk
- WordPress: 6.2-alpha-54678
- Browser: Firefox (108.0.2)
- OS: Windows 10 Professional
- Screen reader: [NVDA](https://nvaccess.org)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by alexstine** [comment]
CC @annezazu

**Comment by ellatrix** [comment]
Thanks @alexstine! I'm having problems reproducing this:

I create a new post (block based theme).
Focus lands in the title, I type something.
Press Enter to create a paragraph and type something.
I press Shift+Tab and focus moves to the block toolbar (block switcher for paragraph).
For you it moves to Editor level options?

I'm testing with Chrome and Firefox, but maybe this is different on Windows.

**Comment by alexstine** [comment]
@ellatrix I will try to record a video of this soon but not sure when I will have time. Out traveling. Maybe this weekend or Monday. Somehow, the options button does receive focus but it is only those specific conditions that make it happen. Seems like this does not happen with an existing post.

**Comment by t-hamano** [comment]
I tested on Windows and could not reproduce the problem. When the paragraph block has focus, if I press `Shift` + `Tab` key, the focus is moved to the block toolbar as intended. If I press the `Shift` + `Tab` key again when the block toolbar has focus, the options button gets focused, but I think this is expected behavior.

For some reason, does it jump over the focus of the block toolbar?

https://user-images.githubusercontent.com/54422211/213659998-a820ed13-a768-4ea4-994d-2dd9c8fe5c28.mp4

**Comment by alexstine** [comment]
This seems to be resolved. Maybe I found it on an older version of trunk or something but I can no longer reproduce it. Very weird. I will still keep an eye on this though as I am not convinced it is not a problem. Closing for now. Hopefully it was just a one-off bug that I caught randomly.
